### DISCUSSION TITLE
Set user as verified automatically

### DISCUSSION THREAD

**Discussion by zicho** [pocketbase/pocketbase#3786] [Q&A] [answered]
So I can't find any way to do this. I want users to be able to log in directly after registration without a confirmation email. I can't find any way to remove this constraint in the admin dashboard. So I tried doing it via this code:

```
const user = await pb.collection("users").create({
    username,
    password,
    passwordConfirm: password,
    verified: true,
});
```

But this throws error (in PB logs):   "errorMessage": "verified: must be a valid value."

But what else is reasonable to put there besides a boolean? Updating any other field works fine. I have tried setting it to "true" (with the quotation marks) and also "1", but same issue.

Is it simply not possible to achieve this using PocketBase?

**NOTE: I have (for testing purposes) removed all API rules for my "users" collection**

**Comment by ganigeorgiev** [comment, marked as answer]
The default validation error message is kindof confusing but regular users are not allowed to change their `verified` status directly. It's value will change to `true` once the user verify their email OR when an admin update it manually.

With that said, the `verified` field doesn't restrict automatically users to authenticate and it's up to the developers to decide whether they want to allow unverified users access to their collections using the API rules (for example `@request.auth.verified = true && ...`).

In other words, if your app doesn't require checking the `verified` user state, you can safely ignore it (_in future versions there will be more control over the builtin system fields and eventually allowing to remove it completely_).

Alternatively, if you really want to set the value to `true` by default, you can register a before create hook. Here is an example using the [JS pb_hooks](https://pocketbase.io/docs/js-overview/):

```js
// pb_hooks/main.pb.js
onRecordBeforeCreateRequest((e) => {
    // skip if admin 
    if (e.httpContext.get("admin")) {
        return;
    }

    e.record.set("verified", true)
}, "users")
```

_edit: If you are using v0.23+ see https://github.com/pocketbase/pocketbase/issues/6051#issuecomment-2525558755._

**Reply by lukepighetti** [reply]
does a user 'verify their email' when they upgrade from a username/password auth to oauth?

**Reply by ganigeorgiev** [reply]
It depends on the result of the linked OAuth2 provider response. In order an existing user to be considered `verified` after OAuth2 auth they need to satisfy one of the conditions:

- the record doesn't have an email
- the email of the record must match with the OAuth2 user email (if returned)
